src/index.js

```diff
@@ -62,6 +85,59 @@ function ContextProvider(props) {
 	return cloneElement(children, rest);
 }
 
+function cloneElementsToShadowRoot(shadowRoot, elements) {
+	elements.forEach((el) => shadowRoot.appendChild(el.cloneNode(true)));
+}
+
+function getAllStyles(target, selector, filter) {
+	const elements = Array.prototype.slice.call(
+		target.querySelectorAll(selector)
+	);
+
+	return filter ? elements.filter(filter) : elements;
+}
+
+const beginInjectingGlobalStyles = (shadowRootRef, injectGlobalStyles) => {
+	cloneElementsToShadowRoot(
+		shadowRootRef,
+		getAllStyles(
+			injectGlobalStyles.target,
+			injectGlobalStyles.selector,
+			injectGlobalStyles.filter
+		)
+	);
+
+	return observeStyleChanges(
```

Comment:
Inlining this function here saves about 23B . It allows us to get rid of the `callback` function wrapping and additional function call as we can just call `cloneElementsToShadowRoot`.